concrete/dense_directed_matrix.go

```diff
@@ -112,8 +115,16 @@ func (g *DirectedDenseGraph) HasEdgeFromTo(u, v graph.Node) bool {
 	return uid != vid && !isSame(g.mat.At(uid, vid), g.absent)
 }
 
-func (g *DirectedDenseGraph) Weight(e graph.Edge) float64 {
-	return g.mat.At(e.From().ID(), e.To().ID())
+func (g *DirectedDenseGraph) Weight(x, y graph.Node) (w float64, ok bool) {
+	xid := x.ID()
+	yid := y.ID()
+	if xid == yid {
+		return 0, true
```

Comment:
The significant of ok is that the weight is known. The weight of an edge between a node and itself here is 0. This may need to be configurable - possibly 1/g.absent?